### PR TITLE
Add preview for mail attachment

### LINKED ISSUE #5701: Ability to directly view attachment from email
Currently it is only possible to either download attachments or save them to files:
image

It would be much nicer if you could just tap on it directly to open it in a viewer to see it directly. :) @ChristophWurst as mentioned

### PR DESCRIPTION
**Pull request by GretaD**
fixes #5701 

- [x] fit the full view on the screen, no scrolling
- [x] fit the mobile view on the screen, no scrolling

### CODE REVIEW

**Comment by ChristophWurst** [comment]
* [ ] BUG the modal is shown as soon as just one of the attachments is a picture. It should only be available to an attachment that is an image. Nothing else.

Bildschirmfoto von 2021-12-27 11-23-24

^ tested with pic & md file

**Comment by ChristophWurst** [comment]
* [ ] BUG images are cropped. Try sending a portrait mode image. It will show landscape with the bottom cropped off

**Comment by GretaD** [comment]
> * [ ]  BUG images are cropped. Try sending a portrait mode image. It will show landscape with the bottom cropped off

i cannot reproduce this one, but im assuming its because the size used to be isMobile: full ? large. I changed it into large only and added overflow scroll. Can you please test, or send me that image you could reproduce

**Comment by GretaD** [comment]
> > * [ ]   BUG images are cropped. Try sending a portrait mode image. It will show landscape with the bottom cropped off
> 
> i cannot reproduce this one, but im assuming its because the size used to be isMobile: full ? large. I changed it into large only and added overflow scroll. Can you please test, or send me that image you could reproduce

i can reproduce and the scrolling fixed it

File: src/components/AttachmentImageViewer.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+<script>
+import Modal from '@nextcloud/vue/dist/Components/Modal'
+export default {
+	name: 'AttachmentImageViewer',
+	components: {
+		Modal,
+	},
+
+	mixins: [isMobile],
```
Comment (GretaD) [inline src/components/AttachmentImageViewer.vue:19]: leftover

**Comment by GretaD** [comment]
desktopatt

mobile
mobileatt

**Comment by ChristophWurst** [comment]
> desktopatt
> 
> mobile mobileatt

look at the pictures. they are still cropped.

**Review by st3iny — COMMENTED** [review]
The overflow of tall pictures is still not fixed. I manually added `overflow: auto` to `.modal-container` and it worked fine (showed a vertical scroll bar).

File: src/components/AttachmentImageViewer.vue [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+::v-deep .modal-wrapper {
+	&--large .modal-container {
+		box-shadow: none !important;
+		display: flex !important;
+	}
+	&--full .modal-container {
+		box-shadow: none !important;
+		display: flex !important;
+		width: 90vw;
+	}
```
Comment (st3iny) [inline src/components/AttachmentImageViewer.vue:39]:
```suggestion
	}
	.modal-container {
		overflow: auto !important;
	}
```

**Comment by GretaD** [comment]
> The overflow of tall pictures is still not fixed. I manually added `overflow: auto` to `.modal-container` and it worked fine (showed a vertical scroll bar).

the thing is that it should fit the screen and not scroll. I fixed it with scrolling but than i checked how viewer does and it doesnt scroll.

**Comment by GretaD** [comment]
please give it a final test. the mobile version of preview its a bit small but readable. I cannot find a solution to fit the screen for portrait and not brake images that are horizontally long. E.g
testhorixontalll

**Review by st3iny — APPROVED** [review]
Images inside modals were still cropped but I deployed a fix.

**Pull request merged**